### PR TITLE
Adding Unit Tests for dl_eval.py

### LINKED ISSUE #848: Unit tests for dl_eval
Write unit tests for the `compute_correct` and `compute_accuracy` functions in `dl_eval.py`. You can approach writing these quick unit tests using `pytest` and having "dummy tensors". You can take inspiration from the `test_loss_function.py` setup-wise of the tensors

### PR DESCRIPTION
**Pull request by codingwithsurya**
This pull request introduces unit tests for the compute_correct and compute_accuracy functions. These tests utilize the pytest framework and incorporate "dummy tensors" to simulate various input conditions.

# Adding Unit Tests for compute_correct and compute_accuracy Functions in dl_eval.py

**What user problem are we solving?**

The current lack of unit tests for the compute_correct and compute_accuracy functions in dl_eval.py presents a potential risk of undetected bugs or errors. By addressing this issue, we aim to ensure the accuracy and reliability of these functions, leading to improved overall code quality and performance.

**What solution does this PR provide?**

This pull request introduces unit tests for the compute_correct and compute_accuracy functions. These tests utilize the pytest framework and incorporate "dummy tensors" to simulate various input conditions. By evaluating different scenarios and edge cases, we can confidently assert the correctness of the functions and identify any potential issues.

**Testing Methodology**
How did you test your changes and verify that existing 
functionality is not broken

During the testing process, I executed the unit tests from the command line interface and verified that all test cases passed without any failures or errors. This confirms the correctness of the `compute_correct` and `compute_accuracy` functions in `dl_eval.py`.

**Any other considerations**
N/A

### CODE REVIEW

File: tests/test_dl_eval.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    predicted = torch.tensor([[0.1, 0.2, 0.7], [0.3, 0.4, 0.3]])
+    actual = torch.tensor([2, 1])
+
+    # Compute the number of correct predictions
+    correct = compute_correct(predicted, actual)
+
+    # Check if the number of correct predictions is correct
+    assert correct == 2
+
+
```
Comment (karkir0003) [inline tests/test_dl_eval.py]: have a test case to verify case of 0 accuracy, maybe some accuracy between 0 and 1 just to verify that the calculation is correct. @yeaitsurya

**Review by karkir0003 — CHANGES_REQUESTED** [review]
add extra unit tests to handle other edge cases

File: tests/test_dl_eval.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    # Check if the accuracy is correct
+    assert accuracy == 1
+
+    # Test case: Accuracy of 0
+    predicted_none_correct = torch.tensor([[0.9, 0.1, 0.0], [0.7, 0.2, 0.1]])
+    actual_none_correct = torch.tensor([2, 1])
+    accuracy_none_correct = compute_accuracy(
+        predicted_none_correct, actual_none_correct
+    )
```
Comment (karkir0003) [inline tests/test_dl_eval.py]: try using parametrized testing to test different cases 

see these docs: https://www.tutorialspoint.com/pytest/pytest_parameterizing_tests.htm

**Review by karkir0003 — CHANGES_REQUESTED** [review]
parametrized testing please

**Comment by codingwithsurya** [comment]
> parametrized testing please

@karkir0003 just commited new changes implementing this. Lmk if its good.

**Review by karkir0003 — APPROVED** [review]
